Dockerfile

```diff
@@ -1,86 +1,381 @@
-FROM circleci/node:14
-RUN sudo npm install -g npm@7
-RUN npm -v
-WORKDIR /tmp
-RUN sudo apt-get update
-RUN sudo apt-get install -y \
-  xdg-utils \
-  libatk-bridge2.0-0 \
-  libgtk-3.0 \
-  libasound2 \
-  lsof \
-  sudo \
-  tcl \
-  expect \
-  zip \
-  lsof \
-  jq \
-  groff \
-  python \
-  python-pip \
-  libpython-dev \
-  less \
-  tree
-
-RUN sudo pip install awscli
+FROM public.ecr.aws/ubuntu/ubuntu:22.04 AS core
+
+ARG DEBIAN_FRONTEND="noninteractive"
+
+# Install git, SSH, and other utilities
+RUN set -ex \
+    && echo 'Acquire::CompressionTypes::Order:: "gz";' > /etc/apt/apt.conf.d/99use-gzip-compression \
+    && apt-get update \
+    && apt install -y -qq apt-transport-https gnupg ca-certificates \
+    && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF \
+    && apt-get install software-properties-common -y -qq --no-install-recommends \
+    && apt-add-repository -y ppa:git-core/ppa \
+    && apt-get update \
+    && apt-get install git=1:2.* -y -qq --no-install-recommends \
+    && git version \
+    && apt-get install -y -qq --no-install-recommends openssh-client \
+    && mkdir ~/.ssh \
+    && mkdir -p /codebuild/image/config \
+    && touch ~/.ssh/known_hosts \
+    && ssh-keyscan -t rsa,dsa -H github.com >> ~/.ssh/known_hosts \
+    && ssh-keyscan -t rsa,dsa -H bitbucket.org >> ~/.ssh/known_hosts \
+    && chmod 600 ~/.ssh/known_hosts \
+    && apt-get install -y -qq --no-install-recommends \
+          apt-utils asciidoc autoconf automake build-essential bzip2 \
+          bzr curl dirmngr docbook-xml docbook-xsl dpkg-dev \
+          e2fsprogs expect fakeroot file g++ gcc gettext gettext-base \
+          groff gzip iptables jq less libapr1 libaprutil1 \
+          libargon2-0-dev libbz2-dev libc6-dev libcurl4-openssl-dev \
+          libdb-dev libdbd-sqlite3-perl libdbi-perl libdpkg-perl \
+          libedit-dev liberror-perl libevent-dev libffi-dev libgeoip-dev \
+          libglib2.0-dev libhttp-date-perl libio-pty-perl libjpeg-dev \
+          libkrb5-dev liblzma-dev libmagickcore-dev libmagickwand-dev \
+          libmysqlclient-dev libncurses5-dev libncursesw5-dev libonig-dev \
+          libpq-dev libreadline-dev libserf-1-1 libsodium-dev libsqlite3-dev libssl-dev \
+          libsvn1 libsvn-perl libtcl8.6 libtidy-dev libtimedate-perl \
+          libtool libwebp-dev libxml2-dev libxml2-utils libxslt1-dev \
+          libyaml-dev libyaml-perl llvm locales make mlocate \
+          netbase openssl patch pkg-config procps python3-configobj \
+          python3-openssl rsync sgml-base sgml-data \
+          tar tcl tcl8.6 tk tk-dev unzip wget xfsprogs xml-core xmlto xsltproc \
+          libzip-dev vim xvfb xz-utils zip zlib1g-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+ENV LC_CTYPE="C.UTF-8"
```

Comment:
PKG_CACHE binaries did not require any changes - they still work in the new image